### PR TITLE
Implement patch proposal timeline

### LINKED ISSUE #134: Patch Proposal Activity - Listing in Editor Tab
Similar to the GitHub plugin, this tab shows:

### Header

* The title of the Patch Proposal 
* The revision number and identifier
* the status of the PP (open, merged, closed, etc.)  
* the author of the PP - and their alias, if I have stored one.
* the datetime when the PP was opened.

### Conversation / Activity: 

* the first entry in the conversation view is always the description added at the time the Patch was opened. 
* the following entries in the conversation view are the comments and/or additional commits, **sorted by the timestamp when they were created* **, as that is returned by the API 
* each comment entry should contain: 
  * `$author_alias (did:...) commented on $date` on the first line
  * the actual comment
  * (no support for suggestions, etc. as of right now)
* each new entry for commits should contain:   
  * `$author_alias (did:...)  commit title` 

We do not yet need to support "Discussions" (that can be resolved/unresolved) as we would need support for that in the underlying model). 

* Note: there is a known issue with this approach: some times "older" comments (in terms of date) - that were written while their author was not connected to the internet -  may appear in the discussion **before** comments that were written **later** (simply because our node didn't know about these comments due to the author having previously been offline). 

This issue is for the initial listing **only**. Actions will be added in separate issues.

Some screenshots from the GitHub plugin follow: 

Image

Image

### PR DESCRIPTION
**Pull request by Stelios123**
closes #134

### CODE REVIEW

File: src/main/java/network/radicle/jetbrains/radiclejetbrainsplugin/patches/PatchProposalPanel.java [hunk 1]
```diff
@@ -225,7 +225,7 @@ protected SimpleChangesBrowser getChangesBrowser(Project project, PatchTabContro
 
     protected void calculatePatchChanges(CountDownLatch latch) {
         ApplicationManager.getApplication().executeOnPooledThread(() -> {
-            // TODO fetch first and then show the changes
+            // TODO fetch first and then show the changes, maybe i have to add the did to the remotes
```
Comment (JChrist) [inline src/main/java/network/radicle/jetbrains/radiclejetbrainsplugin/patches/PatchProposalPanel.java:228]: I think this should be moved as a new issue

File: src/main/java/network/radicle/jetbrains/radiclejetbrainsplugin/models/RadPatch.java [hunk 2]
```diff
@@ -39,7 +39,7 @@ public RadPatch(
 
     public record Revision(
             String id, String description, String base, String oid, List<String> refs,
-            List<Merge> merges, Instant timestamp, List<Discussion> discussions, List<String> reviews) { }
+            List<Merge> merges, Instant timestamp, List<Discussion> discussions, List<Object> reviews) { }
```
Comment (JChrist) [inline src/main/java/network/radicle/jetbrains/radiclejetbrainsplugin/models/RadPatch.java:42]: I guess this is going to be implemented with a proper class when we add reviews?

**Review by JChrist — APPROVED** [review]

File: src/main/java/network/radicle/jetbrains/radiclejetbrainsplugin/patches/PatchProposalPanel.java (hunk 1 above)
Comment (Stelios123) [reply src/main/java/network/radicle/jetbrains/radiclejetbrainsplugin/patches/PatchProposalPanel.java:228]: Yes, I didn't want to make the changes to this branch. Too much changes

File: src/main/java/network/radicle/jetbrains/radiclejetbrainsplugin/models/RadPatch.java (hunk 2 above)
Comment (Stelios123) [reply src/main/java/network/radicle/jetbrains/radiclejetbrainsplugin/models/RadPatch.java:42]: Yes

**Pull request merged**